querier.go

```diff
@@ -221,6 +221,36 @@ func (r *postingsReader) Select(ms ...labels.Matcher) (Postings, []string) {
 	return p, absent
 }
 
+// findTuplesByPrefix uses binary search to find prefix matches within ts.
+func findTuplesByPrefix(m *labels.PrefixMatcher, ts StringTuples) (res []string, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = r.(error)
+		}
+	}()
+	tslen := ts.Len()
+	i := sort.Search(tslen, func(i int) bool {
+		vs, err := ts.At(i)
+		if err != nil {
+			panic(fmt.Errorf("Failed to read tuple %d/%d: %v", i, tslen, err))
```

Comment:
Instead of panicking here, can we set the error and return `true`? This is for two reasons:

1. We almost will NEVER return error.
2. If its an error, we will return immediately after the search, not caring about the result.

I am not sure panic + defer is a good option here.